js/ui/components/networkAgent.js

```diff
@@ -123,11 +130,21 @@ const NetworkSecretDialog = new Lang.Class({
                 secret.valid = true;
 
             if (rtl) {
-                layout.attach(secret.entry, 0, pos, 1, 1);
-                layout.attach(label, 1, pos, 1, 1);
+                let i = 0;
+                if (!this._inputSourceIndicator) {
+                    this._addKbdLayoutSwitcher();
```

Comment:
I would simply call `this._addKbdLayoutSwitcher()` (or even just run the code that is now inside that function) before this loop, and remove the conditionals in here.

Reply:
Yeah, I thought about that one too. I used "this._inputSourceIndicator" to know if we already have added a switcher in the case when we have n entry fields. I am using the loop variable "i" now to only add one switcher for n pw entries.